### PR TITLE
Treat __STDLIB_INTERNAL like strict instead of true

### PR DESCRIPTION
**Pull request by aisamanra**
### Motivation
This will catch regressions from #2054 and #2061 and in general will force a stricter set of standards on our stdlib RBI files. Currently WIP because I'm running this by everyone to make sure we're all okay with it, but it's a pretty simple change.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
just occurred to me: is this going to require that that we must put sigs on methods in `__STDLIB_INTERNAL` RBIs? How does that work if two RBIs add a definition for the same method? It used to be the case that always the sig'd one will win, but now idk... is it like the "second" one?

**Comment by aisamanra** [comment]
It probably will require that every method has a sig. I don't think there's a place in the stdlib RBIs right now where we define methods without sigs, so it didn't matter in that test (although there's another test failure and I haven't dug into that one) but we can if we want special-case `__STDLIB_INTERNAL` to also allow sigless methods but otherwise treat it like `strict`, if we'd like. (That's already how we allow overloading and certain type members on classes, after all, by special-casing specific errors at `__STDLIB_INTERNAL`.)

File: rbi/stdlib/etc.rbi [hunk 1]
```diff
@@ -564,6 +564,10 @@ end
 class Etc::Group < Struct
   extend T::Generic
   Elem = type_member(:out, fixed: T.untyped)
+  class << self
+    extend T::Generic
+    Elem = type_member(fixed: T.untyped)
+  end
 end
```
Comment (aisamanra) [inline rbi/stdlib/etc.rbi]: It turns out that `Etc::Group` both `extend`s and `include`s `Enumerable`, which is why we needed to modify this RBI.

**Comment by elliottt** [comment]
Have you tested this on the mono-repo already?

**Comment by aisamanra** [comment]
Yup: no changes to the monorepo are needed.

**Pull request merged**